tests/components/panasonic_viera/test_remote.py

```diff
@@ -0,0 +1,84 @@
+"""Test the Panasonic Viera remote entity."""
+
+from homeassistant.components.panasonic_viera.const import ATTR_UDN, DOMAIN
+from homeassistant.components.remote import (
+    ATTR_COMMAND,
+    DOMAIN as REMOTE_DOMAIN,
+    SERVICE_SEND_COMMAND,
+)
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_TURN_OFF, SERVICE_TURN_ON
+
+from .test_init import (
+    MOCK_CONFIG_DATA,
+    MOCK_DEVICE_INFO,
+    MOCK_ENCRYPTION_DATA,
+    get_mock_remote,
+)
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+async def setup_panasonic_viera(hass):
+    """Initialize integration for tests."""
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=MOCK_DEVICE_INFO[ATTR_UDN],
+        data={**MOCK_CONFIG_DATA, **MOCK_ENCRYPTION_DATA, **MOCK_DEVICE_INFO},
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    mock_remote = get_mock_remote()
+
+    with patch(
+        "homeassistant.components.panasonic_viera.Remote",
+        return_value=mock_remote,
+    ):
```

Comment:
I'd extract this into a pytest fixture that yields within the patch context manager. That way we don't need to repeat this in every test.

```py
@pytest.fixture(name="mock_remote")
def mock_remote_fixture():
    """Mock the remote."""
    mock_remote = get_mock_remote()

    with patch(
        "homeassistant.components.panasonic_viera.Remote",
        return_value=mock_remote,
    ):
        yield mock_remote
```